microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,587 @@
+        timeout=timeout,
+    )
+    return is_ready    
+
+
+@TestSuiteMetadata(
+    area="power",
+    category="functional",
+    description="""
+    """
+)
+class power(TestSuite):
+    REBOOT_TIME_OUT = 300
+
+    @TestCaseMetadata(
+        description="""
+        This test can be performed in Azure and Hyper-V both. But this script only covers Azure.
+        1. Prepare swap space for hibernation
+        2. Update the grub.cfg with resume=UUID=xxxx where is from blkid swap disk
+        3. Set RTC to future time or past time
+        4. Hibernate the VM
+        5. Resume the VM
+        6. Read RTC timestamp and compare to the original value
+        """,
+        priority=4
+    )
+    def power_clock_sync_hibernate(self, environment: Environment, node: Node, log: Logger) -> None:     
+        # Prepare the swap space in the target VM
+        resource_group = environment.get_information()['resource_group_name']
+        data_disk_name = 'lisa_data_disk_0'
+        disks = node.features[Disk]
+        pattern = re.compile(r"/dev/disk/azure/scsi[1-9]/lun[0-9][0-9]?", re.M)
+        cmd_result = node.execute(
+            "ls -d /dev/disk/azure/scsi*/*", shell=True, sudo=True
+        )
+        matched = find_patterns_in_lines(cmd_result.stdout, [pattern])
+        matched_disk_array = set(matched[0])
+        disk_array: List[str] = [""] * len(matched_disk_array)
+        for disk in matched_disk_array:
+            # readlink -f /dev/disk/azure/scsi1/lun0
+            # /dev/sdc
+            cmd_result = self._node.execute(
+                f"readlink -f {disk}", shell=True, sudo=True
+            )
+            disk_array[int(disk.split("/")[-1].replace("lun", ""))] = cmd_result.stdout
+        if disk_array and disk_array[0] == data_disk_name:
+            log.info("Data disk has already been added")
+        else:
+            disks.add_data_disk(1, schema.DiskType.StandardSSDLRS, 1024)
+
+        dmesg = node.tools[Dmesg]
+        root_msg = re.findall('root=', dmesg.get_output(force_run=True), re.IGNORECASE)
+        env_setup_result = [res for res in root_msg if 'resume' in res]
+        log.debug("Env Setup Result: " + ''.join(env_setup_result))
+        if env_setup_result:
+            log.info("Evironment has been set up...")
+        else:
+            # Configuration for hibernation
+            node.execute(
+                "umask 022"
+            )
+
+            # GetGuestGeneration
+            exit_code = node.execute(
+                "ls /sys/firmware/efi/"
+            ).exit_code
+            os_generation = 2 if exit_code == 0 else 1
+            
+            if isinstance(node.os, Redhat):
+                min_support_kernel = "4.18.0-202"
+                check_vm_support_res = check_vm_feature_support_status(min_support_kernel)
+                current_kernel_version = node.execute(
+                    "uname -r", shell = True, sudo = True
+                )
+                assert_that(
```

Comment:
If it needs to skip the test case, use "SkippedException". The assert failure is failed result, not skipped.